posthog/hogql_queries/insights/trends/test/test_data_warehouse_query_builder.py

```diff
@@ -0,0 +1,221 @@
+from datetime import datetime
+from freezegun import freeze_time
+from posthog.hogql.modifiers import create_default_modifiers_for_team
+
+from posthog.hogql.query import execute_hogql_query
+from posthog.hogql.timings import HogQLTimings
+from posthog.hogql_queries.insights.trends.data_warehouse_trends_query_builder import DataWarehouseTrendsQueryBuilder
+from posthog.hogql_queries.utils.query_date_range import QueryDateRange
+from posthog.schema import (
+    BreakdownFilter,
+    BreakdownType,
+    DateRange,
+    DataWarehouseNode,
+    TrendsQuery,
+)
+from posthog.test.base import BaseTest
+from posthog.warehouse.models import DataWarehouseTable, DataWarehouseCredential
+
+from boto3 import resource
+from botocore.config import Config
+from posthog.settings import (
+    OBJECT_STORAGE_ACCESS_KEY_ID,
+    OBJECT_STORAGE_BUCKET,
+    OBJECT_STORAGE_ENDPOINT,
+    OBJECT_STORAGE_SECRET_ACCESS_KEY,
+)
+import s3fs
+from pyarrow import parquet as pq
+import pyarrow as pa
+
+from posthog.test.base import (
+    ClickhouseTestMixin,
+    snapshot_clickhouse_queries,
+)
+from posthog.hogql_queries.legacy_compatibility.filter_to_query import (
+    clean_entity_properties,
+)
+
+TEST_BUCKET = "test_storage_bucket-posthog.hogql.datawarehouse.trendquery"
+
+
+class TestDataWarehouseQueryBuilder(ClickhouseTestMixin, BaseTest):
+    def teardown_method(self, method) -> None:
+        s3 = resource(
+            "s3",
+            endpoint_url=OBJECT_STORAGE_ENDPOINT,
+            aws_access_key_id=OBJECT_STORAGE_ACCESS_KEY_ID,
+            aws_secret_access_key=OBJECT_STORAGE_SECRET_ACCESS_KEY,
+            config=Config(signature_version="s3v4"),
+            region_name="us-east-1",
+        )
```

Comment:
Does this talk to S3 over the web? Or is there some local magic going on?

Reply:
Ah yeah forgot to note that this works locally when minio is up

Reply:
tests working now. was using bad configs